File: SecondYear/adityshrma08_AdityaSharma_2226csit1014_2/Week_3/Binary Search.cpp
int search(vector<int> &nums, int target) {
    // Write your code here.
    int s=0,e=nums.size()-1;
    int mid;
    while(s<=e){
    mid = (s+e)/2;
    if(nums[mid]==target){
        return mid;
    }
    else if(nums[mid]>target){
        e = mid-1;
    }
    else{
        s = mid + 1;
    }
    }
    return -1;
}

File: SecondYear/Anantj2499_AnantJaiswal_2226cseml1088_2/Week_3/BinarySearch.java
public class BinarySearch {
    public static int search(int []nums, int target) {
        // Write your code here.
        int start =0;
        int end = nums.length-1;
        int pos = -1;
        while(start<=end){
            int mid = (start+end)/2;
            if(nums[mid]==target){
                pos = mid;
                break;
            }
            if (nums[mid]<target)
                start=mid+1;
            if (nums[mid]>target)
                end = mid-1;
        }
        return pos;
    }
}